sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -523,6 +531,30 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) with Logging {
     super.dialectClassName
   }
 
+  override protected[sql] def newOptimizer: Optimizer = new Optimizer {
```

Comment:
Can we override the `lazy val optimizer: Optimizer` directly? So we don't need to change any code in `SQLContext`